src/Components/ProductsPage/ProductsPage.tsx

```diff
@@ -0,0 +1,139 @@
+import React, {
+  useContext, useEffect, useMemo, useState,
+} from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { deductDiscount } from '../../Helpers/functions/deductDiscount';
+import { Product } from '../../Helpers/types/Product';
+import { Quantity } from '../../Helpers/types/Quantity';
+import { Sorter } from '../../Helpers/types/Sorter';
+import { SearchPage } from '../../Pages/SearchPage/SearchPage';
+import { Breadcrumbs } from '../Breadcrumbs/Breadcrumbs';
+import { QueryContext } from '../Context/QueryContext';
+import { NoResults } from '../NoResults';
+import { Pagination } from '../Pagination/Pagination';
+import { ProductsList } from '../ProductsList/ProductsList';
+import { Select } from '../Select/Select';
```

Comment:
Will be better if you create index files and import all of this from components folder